Created base RLlib documentation (#635)
huawei-noah/SMARTS: ultra/docs/rllib.md

# RLlib support in ULTRA

## Summary

> We have integrated Ray's Reinforcement Learning framework inside ULTRA to cater users whom are familiar with RLlib. We have blended ULTRA's environment and agent specifications 

Reviewer: ```suggestion
We have integrated Ray's Reinforcement Learning framework inside ULTRA to cater to users who are familiar with RLlib. We have blended ULTRA's environment and agent specifications 
```

> inside RLlib's configuration to allow their policies to interact in our environment. We currently support RLlib's PPO policy and the plan is to add provision for SAC, DDPG and DQN.

Reviewer: ```suggestion
inside RLlib's configuration to allow their policies to interact with the ULTRA environment. We currently support RLlib's PPO policy, but plan to support SAC, DDPG and DQN.
```

## Setup

> To use the RLlib functionality inside ULTRA, you must have the ultra-rl>=0.2. If you have not followed the setup guide for ULTRA, here is a [link](docs/setup.md) to it 

Reviewer: ```suggestion
To use RLlib with ULTRA, you must have the ultra-rl>=0.2. If you have not followed the setup guide for ULTRA, here is a [link](docs/setup.md) to it 
```

of the steps is shown below

1) To generated maps used by the ULTRA scenarios, run the following command: 
  ```sh
  $ scl scenario build-all ultra/scenarios/pool
  ```
2) For building scenarios from specified task configurations, run the following command
  ```sh
  $ python ultra/scenarios/interface.py generate --task <TASK> --level <LEVEL>
  ```
3) Execute `ultra/rllib_train.py`. The following is a list of available arguments.
  - `--task`: The task number to run (default is 1).
  - `--level`: The level of the task (default is easy).
  - `--episodes`: The number of training episodes to run (default is 100).
  - `--timestep`: The environment timestep in seconds (default is 0.1).
  - `--headless`: Whether to run training without Envision (default is False).
  - `--eval-episodes`: The number of evaluation episodes (default is 200).
  - `--eval-rate`: The rate at which evaluation occurs based on the number of episodes (default is 10000).
  - `--seed`: The environment seed (default is 2).
  - `--policy`: The policy (agent) to train (default is ppo). Only PPO is supported for now.
  - `--log-dir`: The directory to put models, tensorboard data, and training results (default is logs/).

>   - `--max-samples` : The number of samples per iteration (default is 4000)

Reviewer: max-samples is unclear to me, looking at kimia's PR it looks like its the training batch size? We should probably change the arg name or doc string, but I suppose that comment should be  on the other PR.

Author: Okay, it seems that I had "max-samples" argument when I was doing my tests, yes you are correct, I have changed it to "--training-batch-samples", let me know if you are okay with this :)
